src/fauxtoshop.cpp

```diff
@@ -1,517 +1,487 @@
-            cout << "Saving image as '" << newImageName << "' successful!" << endl;
-            break;
-        }
-        //Reprompt the user if there is an issue with the saving
-        cout << "Oops! Something went wrong... Please try again" << endl;
-    }
-}
+////
+/// This is an implementation of a graphics editing program that allows the user to
+/// edit an image using 4 filters:
+/// a. Scatter - blurs the image
+/// b. Edge detection - creates a black & white image
+/// c. 'Green screen' a sticker image onto a background image
+/// c. Compare an image with another image
+/// After the filter has been applied the user has the option of saving the image
+/// and repeating the filters application on onto the same image or another image
+////
+
+// Using Stanford libraries
+#include "console.h"
+#include "gbufferedimage.h"
+#include "grid.h"
+#include "gwindow.h"
+#include "fauxtoshop-provided.h"   // instructor-provided code
+#include "simpio.h"
+#include "random.h"
+#include "strlib.h"
+#include "vector.h"
+// Using the c++ standard libraries
+#include <iostream>
+using namespace std;
+
+// Function prototypes
+bool nameImageFileToOpen(
+        GBufferedImage &img, const string& prompt, bool exitWhenUserEntersBlank = true);
+void openImage(GBufferedImage &img, GWindow &gw);
+int selectFilterOptions();
+int getValidInteger(const string& promptMessage, int lowerBound, int upperBound);
+void getRandomPixelColor(
+        const Grid<int> &original, Grid<int> &scatterGrid, int row, int col, int scatterRadius);
+void applyScatterFilter(GBufferedImage &img);
+void applyEdgeDetectionFilter(GBufferedImage &img);
+void applyGreenScreenFilter(GBufferedImage &img);
+void compareImages(const GBufferedImage& img, GWindow &gw);
+void saveImage(GBufferedImage &img);
+int calculateRBGColourDifference(int pixel1, int pixel2);
+void validateLocationInput(int& row, int& col);
+
+int main() {
+    // 1. Greet the user
+    cout << "Welcome to Fauxtoshop!" << endl;
+    // Basic setup of Graphics Window
+    GWindow gw;
+    gw.setTitle("Fauxtoshop");
+    gw.setExitOnClose(true);
+    gw.setVisible(true);
+    // Enable the same random numbers on each run
+    fakeRandomNumberGenerator();
+    // Set up for the main image
+    GBufferedImage img;
+
+    // Ask the user to open an image file
+    while (nameImageFileToOpen(
+               img,
+               "Enter name of image file to open ('?' to open file selector or blank to quit): ")){
+        openImage(img, gw);
+        // User confirms which filter options to use: scatter, edge detection, 'green screen',
+        // compare image
+        int optionNum = selectFilterOptions();
+
+        // Complete filtering effect based on filter choice:
+        // 1.scatter, 2. edge detection, 3. 'green screen', 4. compare image
+        switch (optionNum){
+            case 1:
+                applyScatterFilter(img);
+                // Prompt the save to ask if they would like to save the new image
```

Comment:
Nit: I think there might be a slight typo here. Maybe "Ask the user if they would like to save the new image"?